src/SW_netCDF_Input.c

```diff
@@ -0,0 +1,8443 @@
+#include <math.h>                       // for NAN, ceil, isnan
+#include <netcdf.h>                     // for NC_NOERR, nc_close, NC_DOUBLE
+#include <stdio.h>                      // for size_t, NULL, snprintf, sscanf
+#include <stdlib.h>                     // for free, strtod
+#include <string.h>                     // for strcmp, strlen, strstr, memcpy
+
+/* =================================================== */
+/*                   Local Defines                     */
+/* --------------------------------------------------- */
+
+/** Progress status: SUID is ready for simulation */
+#define PRGRSS_READY ((signed char) 0)
+
+/** Progress status: SUID has successfully been simulated */
+#define PRGRSS_DONE ((signed char) 1)
+
+/** Progress status: SUID failed to simulate */
+#define PRGRSS_FAIL ((signed char) -1)
+
+#define NIN_VAR_INPUTS 21
+
+/* Columns of interest, and excludes:
+    - Input key and input name
+    - "do input" flags in value
+    - Input file name/pattern
+    - St years and stride years start
+    - Calendar override
+    - User comment */
+#define NUM_INPUT_INFO 14
+
+/* Maximum number of variables per input key */
+#define SW_INNMAXVARS 22
+
+/* Indices within `inWeathStrideInfo` for stride year and stride start */
+#define SW_INSTRIDEYR 0
+#define SW_INSTRIDESTART 1
+
+static const char *const expectedColNames[] = {
+    "SW2 input group",    "SW2 variable",         "SW2 units",
+    "Do nc-input?",       "ncFileName",           "ncVarName",
+    "ncVarUnits",         "ncDomainType",         "ncSiteName",
+    "ncCRSName",          "ncCRSGridMappingName", "ncXAxisName",
+    "ncYAxisName",        "ncZAxisName",          "ncTAxisName",
+    "ncStrideYears",      "ncStrideStart",        "ncStridePattern",
+    "ncCalendarOverride", "ncVAxisName",          "Comment"
+};
+
+/* This array and `possVarNames` must line up the variables within each key */
+static const char *const swInVarUnits[SW_NINKEYSNC][SW_INNMAXVARS] =
+    {
+        {"1", "1"},                           /* inDomain */
+        {"1", "degree_north", "degree_east"}, /* inSpatial */
+        {"1", "m", "degree", "degree"},       /* inTopo */
+        {"1",        "cm", "cm",   "g cm-3", "cm3 cm-3", "g g-1", "g g-1",
+         "g g-1", /*inSoil*/
+         "cm3 cm-3", "1",  "degC", "1",      "1",        "1",     "1",
+         "1",        "NA", "NA",   "NA",     "NA",       "NA",    "NA"},
+        {"1",      "m2 m-2", "m2 m-2", "g m-2", "g m-2",
+         "1",      "g m-2", /*inVeg*/
+         "m2 m-2", "g m-2",  "g m-2",  "1",     "g m-2",
+         "m2 m-2", "g m-2",  "g m-2",  "1",     "g m-2",
+         "m2 m-2", "g m-2",  "g m-2",  "1",     "g m-2"},
+        {"1", /* inWeather */
+         "degC",
+         "degC",
+         "cm",
+         "%",
+         "m s-1",
+         "m s-1",
+         "m s-1",
+         "%",
+         "%",
+         "%",
+         "%",
+         "degC",
+         "kPa",
+         "m s-1"},
```

Comment:
The hard-coded units for shortWaveRad should be "NA" instead of current "m s-1" (see tsv nc-inputs) -- and the code should skip the check if the units are matching (the same as for the units of the tsv nc-inputs swrcp[*]). This is because different datasets use different approaches for solar radiation (flux density [W m-2] vs. irradiation [MJ m-2]) which we handle via `desc_rsds`.